### PR TITLE
Add fmt based data rendering

### PR DESCRIPTION
**Pull request by ngrodzitski**
###  :notebook_with_decorative_cover: Description

This is a draft PR for review. 

###  :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [ ] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: libvast/src/data.cpp [hunk 1]
```diff
@@ -518,6 +519,10 @@ bool convert(const caf::config_value& x, data& y) {
   return caf::visit(f, x);
 }
 
+std::string to_string(const data& d) {
+  return fmt::format("{:a}", d);
+}
```
Comment (dominiklohmann) [inline libvast/src/data.cpp:524]: We can probably just add a `using fmt::to_string` somewhere in the namespace `vast`. Maybe in `fwd.hpp`.

File: libvast/vast/fmt_integration.hpp [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+  }
+};
+
+template <>
+struct formatter<std::pair<std::string_view, vast::view<vast::data>>>
+  : formatter<vast::record::value_type> {};
+
+/// Definition of fmt-formatting rules for vast::data.
+template <>
+struct formatter<vast::data> : public vast::detail::vast_formatter_base {
```
Comment (dominiklohmann) [inline libvast/vast/fmt_integration.hpp:324]: The formatter for `vast::data` should live in `vast/data.hpp`

**Review by dominiklohmann — COMMENTED** [review]
I like the approach you've taken with the custom base formatter class. I've had some small comments from a first look at this. Feel free to ask for further reviews if something's unclear.

**Comment by dominiklohmann** [comment]
[ch23155]

**Review by dominiklohmann — COMMENTED** [review]
This looks mostly ready to merge to me. Just two open things:

- Remove the commented code
- Rebase onto recent master

**Comment by ngrodzitski** [comment]
Hi @dominiklohmann,
This one is ready for review.

**Comment by dominiklohmann** [comment]
> Hi @dominiklohmann,
> This one is ready for review.

Thanks! I will take a look at this today and we'll likely have it merged by tomorrow.

File: libvast/src/data.cpp (hunk 1 above)
Comment (ngrodzitski) [reply libvast/src/data.cpp:518]: Adding this works bad with concept/printable that is still there.

Comment (dominiklohmann) [reply libvast/src/data.cpp:518]: Yes, this will have to wait until printable is removed entirely.

**Pull request merged**